content/blog/federating-spire-on-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,447 @@
+spire-agent-jhgwf               3/3     Running   0             37d
+spire-agent-sm8gt               3/3     Running   0             37d
+spire-server-574474c7dc-gbzl6   2/2     Running   1 (11d ago)   37d
+```
+
+```shellsession
+Cluster2:~ # kubectl get po -n spire 
+
+NAME                            	 READY      STATUS      RESTARTS       AGE 
+
+spire-agent-wttmd               	 3/3        Running     1 (24h ago)    24h 
+spire-server-574474c7dc-2bfcx   	 2/2        Running     0              24h 
+```
+
+# Step 2. Installing Istio
+
+On each of your Kubernetes clusters, install Istio and patch Istio ingress gateway. Istio can detect the existence of a UNIX Domain Socket that implements the Envoy SDS API on a defined socket path, allowing Envoy to communicate and fetch identities directly from it. SPIRE can be configured for Istio workloads through an integration with Envoy’s SDS API. 
+
+## 2.1 Download the latest release
+
+You can download the latest release using the official Istio repository or just copy the following command, which would do the same thing for you.
+
+```shellsession
+curl -L https://istio.io/downloadIstio | sh - 
+```
+
+Change to the Istio directory (cd command), and set the path by using this command:
+
+```shellsession
+cd istio-1.17.1 
+export PATH=$PWD/bin:$PATH 
+```
+
+## 2.2 I﻿nstall Istio with custom patch
+
+Install Istio with custom patches for the Ingress-gateway as well as for Istio-proxy.  
+
+Get the Istio-spire-config patch using this [link](https://github.com/cxteamtrials/caas-trials-content/blob/main/services/istio/release-1.17/spire), and install that patch using the following commands:
+
+```shellsession
+#For K8s cluster 1:
+istioctl apply -f services/istio/release-1.17/spire/spire-patch-cluster1.yaml 
+#For K8s cluster 2:
+istioctl apply -f services/istio/release-1.17/spire/spire-patch-cluster2.yaml 
+```
+
+Installing Istio with the custom patch will share the spiffe-csi-driver with the Ingress Gateway and the sidecars that are going to be injected on workload pods, granting them access to the SPIRE Agent’s UNIX Domain Socket. 
+
+## 2.3 Patch Istio Ingress Gateway
+
+### 2﻿.3.1 Apply SPIFFE ID
+
+First, you must get and apply one of SPIRE controller manager’s [CRD (Custom Resource Definition)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) ClusterSPIFFEID. The CRD - ClusterSPIFFEID is a cluster-wide resource used to register workloads with SPIRE. The ClusterSPIFFEID can target all workloads in the cluster or can be optionally scoped to specific pods or namespaces via label selectors.  
+
+Create a ClusterSPIFFEID CRD to generate registration entries in SPIRE server for all workloads labeled ***spiffe.io/spire-managed-identity: true.***
+
+Get the ClusterSPIFFEID used for this demo using this [link](https://github.com/cxteamtrials/caas-trials-content/blob/main/services/spire/clusterspiffeid-example.yaml) and  apply it to both clusters. 
```

Comment:
I am not sure I fully understand how one will get the ClusterSPIFFEID using the yaml file https://github.com/cxteamtrials/caas-trials-content/blob/main/services/spire/clusterspiffeid-example.yaml ?

In the sentence "Get the ClusterSPIFFEID used for this demo using this [link](https://github.com/cxteamtrials/caas-trials-content/blob/main/services/spire/clusterspiffeid-example.yaml) and  apply it to both clusters", it sounds to me there are two actions here: 
1- get the ClusterSPIFFIDID 
2- apply the ClusterSPIFFEID to each demo cluster.

However, there is only one command mentioned in the section 2.3.1 Apply SPIFFE ID:
kubectl apply -f services/spire/clusterspiffeid-example.yaml